packages/server/api/types/fastify.d.ts

```diff
@@ -27,8 +29,10 @@ declare module 'fastify' {
         Logger = unknown,
         RequestType = unknown,
     > {
-        principal: ContextConfig extends { allowedPrincipals: infer P extends readonly PrincipalType[] }
-            ? PrincipalForTypes<P>
+        principal: ContextConfig extends { security: AuthenticatedRoute }
+            ? ContextConfig['security']['authorization'] extends { allowedPrincipals: readonly (infer P extends PrincipalType)[] }
+                ? PrincipalForType<P>
+                : Principal
             : Principal
         rawBody?: string | Buffer
```

Comment:
we should have `project` object here and it should be infered depend if the security authorization type is project otherwise it will be never or undefined.